### PR TITLE
feature: update TF framework version support

### PR DESCRIPTION
**Pull request by saimidu**
*Issue #, if available:*

*Description of changes:*
Updated the latest framework versions for TensorFlow on Estimators and Models

*Testing done:*
All tests pass on py36 env when running `tox tests/unit`

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/fw_utils.py [hunk 1]
```diff
@@ -59,7 +59,7 @@
     "local_gpu",
 )
 SM_DATAPARALLEL_SUPPORTED_FRAMEWORK_VERSIONS = {
-    "tensorflow": ["2.3.0", "2.3.1", "2.4.1"],
+    "tensorflow": ["2.3.1", "2.3.2", "2.4.1"],
```
Comment (ChoiByungWook) [inline src/sagemaker/fw_utils.py:62]: Is this removing 2.3.0 as a valid version?

Comment (saimidu) [reply src/sagemaker/fw_utils.py:62]: Correct. TF 2.3.0 images are not usable with SM DataParallel

File: tests/conftest.py [hunk 2]
```diff
@@ -238,20 +238,20 @@ def tensorflow_training_py_version(tensorflow_training_version, request):
...
 
 def _tf_py_version(tf_version, request):
     version = Version(tf_version)
-    if Version("1.15.4") <= version < Version("1.16"):
+    if version == Version("1.15") or Version("1.15.4") <= version < Version("1.16"):
         return "py3"
     if version < Version("1.11"):
         return "py2"
-    if Version("2.0.3") <= version < Version("2.1"):
+    if version == Version("2.0") or Version("2.0.3") <= version < Version("2.1"):
```
Comment (ahsan-z-khan) [inline tests/conftest.py:252]: Why do we removing the patch/revision numbers?

Comment (saimidu) [reply tests/conftest.py:252]: We are not removing the previous patch numbers. I changed it to make the 2.0.0, 2.0.1, and 2.0.2 versions be tested with py2, while 2.0.3 and 2.0.4, which weren't released with py2, are only tested with py3.

Moreover, "2.0" previously pointed to "2.0.1" on the `tensorflow.json` file. I have now updated it to point it to the latest patch version, which means that "2.0" cannot be tested with py2 anymore.

**Review by ajaykarpur — APPROVED** [review]

**Pull request merged**